temba/flows/models.py

```diff
@@ -670,7 +669,9 @@ def handle_actionset(cls, actionset, step, run, msg, started_flows, is_test_cont
 
     @classmethod
     def handle_ruleset(cls, ruleset, step, run, msg, started_flows, resume_parent_run=False, resume_after_timeout=False):
-        if msg.status == INTERRUPTED:  # check interrupt
+        from temba.ussd.models import USSDSession
+
+        if ruleset.is_ussd() and run.session and run.session.status == USSDSession.INTERRUPTED:
```

Comment:
```run.session and run.session.status == USSDSession.INTERRUPTED```

Would be nice to have this encapsulated since I see it 4 times in this PR. Also, isn't it a bit confusing that we now can have run.exit_type of interrupted and run.session.status of interrupted? Are you sure we need both? Are they ever different?